### ISSUE TITLE
Previous version available for Macs? 

### ISSUE THREAD

**Issue by enazario11** [olivierfriard/BORIS#729]
Hello, 

I saw the folder of previous version files linked on the BORIS website, but was not seeing previous version download links available for Mac operating systems. Would you happen to have the Mac download link for version 8.20.3? 

Thank you, 
Emily

**Comment by outergrace** [comment]
Hi,

Replying in this chain because I am having the same issues. I am also not able to download BORIS on Mac OS ventura, and not able to run it through Linux Virtual Machine. Wondering when the new version will be available for Mac OS and if there is an older model I can download in the mean time. 
Thanks,

Grace

**Comment by Kpraetori** [comment]
I also would like this. 
I got the Docker container as an alternate but it won't open and gives me an error about "Qt platform plugin could be initialized" and I don't know how to fix it. I tried the virtual machine as well and got an error with a file folder structure and then the whole VM crashes afterwards. I've got the shell to run from a windows with UTM but can't get past it to boot anything. I'd be happy with an older version as right now, too.

**Comment by AppleForever1984** [comment]
Yeah, not sure why this project has gone so insanely far off the deep end with the macOS hate w/o more explanation?
BORIS supported macOS for YEARS, but I am pretty sure the last macOS release they put out is/was (and apparently ever will be, given the blow-off since by Olivier & co. - not even giving us a desperate last version download link and clear notice of same or some other reason at least?) as follows:
v7.13.9, released on 2022-10-25.

THANKFULLY, I save installers and have a .dmg for ab older system we need to use at present (it still works on Monterey and Ventura so far, and have not needed/tried Sonoma), but yeah, this could have been handled WAAAAAY better after all of these years.  As a macOS user, I TOTALLY Get the IDIOCY of their constant annual releases and quick abandonment of macOS versions after 3 years, but lots of other small devs manage to keep doing it, even if Sonoma was a PiTA that took longer...

That said, SINCEREST THANKS for all the years of support - some projects never even bother at all.
Still, whatever caused the abandonment, the majority of macOS users are not the problem, so we just wish you could do more than say, drop dead or go get Windows/Linux somehow?  Surely, that's not too much to ask as a goodbye?

**Comment by derekreilly** [comment]
This is unfortunate, we really wanted to become a BORIS lab. Why is there a Mac download link on the website?

**Comment by AppleForever1984** [comment]
Apparently, the only reason for the macOS download page is to tell us all to get use Windows/Linux (VM or otherwise)?

It's more than just unfortunate.  Why not just leave us the last macOS version with a caveat that it is unsupported and a quick/rational explanation as to why, after all these years of support?  I've seen other projects explain that maybe they do not have Macs available to compile and test, or maybe the macOS person on the project left, or such.  The last version was there for a while when v8 very first came out, or else I wouldn't have it.  The 7.13.9 version still works JUST FINE (for now), but instead we got inexplicably cutoff at the knees with no explanation for whatever reason.

This site does not allow .dmg uploads to share the last version amongst ourselves, and it'd be a trust issue for anyone else anyway.  More reason why not keeping it up for us on the BORIS site seems a bit over the top, as if there's some unknown animus about macOS someone is not sharing. 🤷‍♂️

**Comment by AppleForever1984** [comment]
I found this site that allows for fre uploads and sharing of files for up to a week for free (free uoploads only area llowed for a week), so if you're feeling trusting, go for it:
[https://www.filemail.com/d/fmuufkkvisbueaf](https://www.filemail.com/d/fmuufkkvisbueaf)

**Comment by kchudeau** [comment]
I would also appreciate some explanation for why Macs are no longer supported and/or providing a previous version with the stipulation that it is unsupported. I have remote undergraduate research assistants who use BORIS on their own computers which are a mixture of PCs and Macs of different version and trying to get everyone running Linux turns our coding meetings into running into tech issue after tech issue similar to @Kpraetori. 

Having this software available outside of a lab setting has been a lifesaver and has been such an inclusive way of getting my undergraduates involved in behavior research!! So despite the small bugs that previous versions may have, giving us something rather than nothing (or very complicated ways to run through VM) I think is preferred. 

I am currently running BORIS v 7.9.7 on my Mac Monterey OS and have relatively few glitches other than zooming issues.

**Comment by olivierfriard** [comment]
Hi,
no need to share installers because (as specified on the BORIS web site):

> All previous versions of BORIS are available in the [BORIS
> releases](https://github.com/olivierfriard/BORIS/releases) archive or in the [BORIS archive](https://drive.google.com/drive/folders/0B0KTzxkLpPJ_bE8zOUFjenFnU3c?resourcekey=0-b9_Ng_8SqK-FIuOP8qIn7w)
> for versions prior to 2.99. Please note that the backward compatibility
> of a BORIS project is not guaranteed!

However, I can confirm that the latest version running natively on macOS is 7.13.9 (https://github.com/olivierfriard/BORIS/releases/tag/v7.13.9).

This version is 2 years old and is no longer maintained. [Many bugs were fixed](https://github.com/olivierfriard/BORIS/wiki/BORIS-change-log-v.8) and [features were added](version8.md) in the current version.

I recently added a paragraph on the [BORIS website](https://www.boris.unito.it/download_mac/) regarding the unavailability of a version for macOS.

**Comment by AppleForever1984** [comment]
> However, I can confirm that the latest version running natively on macOS is 7.13.9 (https://github.com/olivierfriard/BORIS/releases/tag/v7.13.9).

Therein lies the rub!
The macOS page tells us what is **_not_** available, but why not simply also add the above sentence and link to that last available version for macOS as well, since that's what macOS users are actually looking to find?  We already know full well that it is always provided WITHOUT WARRANTY, so a link to v7.13.9 on that same macOS page, right before all of the other **_much less viable and more convoluted_** ways to try to use v8.x seems resonable, no?  The note about archives is buried on the downloads page and does **_not_** say which is the actual last version for macOS (that was clear only for the very first v8 release at that time, which is why I knew, barring going through the archive pages).

That said, thank you **_very much_** for better explaining the situation and helping us confirm that 7.13.9 is/was the last/latest macOS version.  We are running 7.13.9 just fine on Monterey and Ventura, so we've been just fine with same.  **_Yes_**, it has some bugs (**_EVERY_** version has!), but none have been so fatal that we've not continued to use your software effectively, efficiently and with much gratitude.  Thanks again, especially for the additional information.  Hopefully mpv will be resolved soon (the only functioning project I see for that on macOS is iina, and not sure if you've talked to those devs about how they got mpv going on macOS?  I often find that devs are often very amenable to sharing their successes with their fellow devs!).

**Comment by AppleForever1984** [comment]
Thanks again - very much!
Bummer about iina, and mpv.  Maybe one day...

**Comment by AppleForever1984** [comment]
BTW, yes, it does look like [trying to use/test mpv with a VM won't work well, if at all](https://github.com/mpv-player/mpv/issues/12518), since the actual hardware (or driver) is expected.  No Macs around there at all, eh?  Wish we had one to spare/share...but it may be it works fine on bare metal even if it's not in your VM.
